document/workflow/variable-tree/data-source/skeleton-tree-data-source.ts

```diff
@@ -154,8 +159,14 @@ export class SkeletonTreeNode implements GraphicallyRepresented<Element>, Dragga
     }
 
 
-    get allChildrenMarked(): boolean {
-        return this.children.length > 0 && !this.children.some(child => !child.marked);
+    markIfChildrenMarked() {
+        // all children marked
+        if (this.children.length > 0 && !this.children.some(child => !child.marked)) {
+            this.marked = true;
+        }
+        if (!this.parent?.marked) {
```

Comment:
Is it not enough to check parent, if (this.children.length > 0 && !this.children.some(child => !child.marked)) = true?